proposals/NNNN-Keyboard-Enhancements.md

```diff
@@ -0,0 +1,219 @@
+6. HMI should show default characters in the remaining customizable Keys. if _"customizeKeys"_ array is less than or equal to the customizable keys allowed.
+7. If a certain special character is not supported by the system. The HMI should send _"WARNING"_ response, with _"info"_ text as _" some symbols might not be supported by system"_. This will give a chance to use symbols that are supported and also inform the app about the system not supporting certain characters.
+
+**Example usage:** If an app wants to change three symbols, the app sends the value of _"customizeKeys"_ as _"₹£$"_. The parameter follows a similar structure as _"limitedCharacterList"_.
+ 
+ ```xml
+     <struct name="KeyboardProperties" since="3.0">
+        <description>Configuration of on-screen keyboard (if available).</description>         
+		:       
+		<param name="maskInputCharacters" type="KeyboardInputMask" mandatory="false"  since="X.X"> 
+			<description>Allows an app to mask entered chanracters on HMI</description>
+		</param> 
+		<param name="customizeKeys" type="String" maxlength="1" minsize="1" maxsize="10" array="true" mandatory="false"  since="X.X">
+            <description>Array of special characters to show in customisable Keys.</description>
+            <description>If omitted, keyboard will show default special characters</description>
+        </param>  
+    </struct>
+ ```
+
+#### Change in _"RegisterAppInterface"_ response
+ 
+```xml
+<function name="RegisterAppInterface" functionID="RegisterAppInterfaceID" messagetype="response" since="1.0">
+		:
+        <param name="keyboardCapabilities"  type="KeyboardCapabilities" mandatory="false" since="X.X">
+            <description>See KeyboardCapabilities</description>
+        </param>
+</function>
+```
+
+## Alternatives considered 
+
+Instead of returning _"KeyboardCapabilities"_ in _"RegisterAppInterface"_, struct _"ScreenCapability"_ can be used. [Struct _"ScreenCapability"_ is defined in Widget Support Proposal.](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0216-widget-support.md)
+
+### Change in ScreenCapability
+
+```xml
+<struct name="ScreenCapability" since="5.x">
+    :
+    <param name="keyboardCapabilities"  type="KeyboardCapabilities" mandatory="false" since="X.X">
+        <description>See KeyboardCapabilities</description>
+    </param>
+</struct>
+```
+
+## HMI changes
+
+1. The HMI should add the support for numeric keypad.
+2. In cases, where App sets _"maskInputCharacters"_ as _"USER_CHOICE_INPUT_KEY_MASK"_, HMI Keyboard should provide a button to the user to show/hide characters entered using the keyboard. 
+3. In cases, where app sets _"maskInputCharacters"_ as _"ENABLE_INPUT_KEY_MASK"_ or _"DISABLE_INPUT_KEY_MASK"_, HMI keyboard will not provide a button to the user to show/hide characters entered using the keyboard. 
+
+## Android and iOS changes

```

Comment:
Are these Android and iOS changes part of the `Alternatives considered` section? If so, please update header to ###.  If not, please move to the `Proposed solution section` also with ### header formatting.

Reply:
Added changes in c845c7115e0d5fa37dbe6ca15e12aceddf02db16